### PR TITLE
Add datasets section and list endpoints

### PR DESCRIPTION
**Pull request by marianmosley**
# What
`Datasets` API doc section and  list functions.
# Why

Part of:
mode/issues#16096
mode/issues#16145
mode/issues#16187

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -129,6 +129,108 @@ paths:
         description: "`Chart` token"
         required: true
         type: string
+  "/{account}/spaces/{space}/datasets":
+    x-summary: Datasets
+    get:
+      operationId: listDatasetsInSpace
+      summary: List datasets for a space
+      description: Returns all `Datasets`s in a `Space`
```
Comment (amirbey) [inline swagger.yaml]:
```suggestion
      description: Returns all `Dataset`s in a `Space`
```

File: swagger.yaml [hunk 2]
```diff
@@ -129,6 +129,108 @@ paths:
...
+        required: true
+        type: string
+      - name: space
+        in: path
+        description: "`Space` token"
+        required: true
+        type: string
+      - name: filter
+        in: query
+        description: 'Returns a filtered list of `Dataset`s in a `Space` filtered
```
Comment (amirbey) [inline swagger.yaml:164]:
```suggestion
        description: 'Returns a filtered list of `Datasets` in a `Space` filtered
```

File: swagger.yaml [hunk 3]
```diff
@@ -129,6 +129,108 @@ paths:
...
+          by whether their `created_at` or `updated_at` timestamps are `gt` (greater
+          than) or `lt` (less than) the datetime passed in (ISO8601 format). Example:
```
Comment (amirbey) [inline swagger.yaml]:
```suggestion
          than) or `lt` (less than) the datetime passed in ISO8601 format. Example:
```

File: swagger.yaml [hunk 4]
```diff
@@ -129,6 +129,108 @@ paths:
...
+          `?filter=created_at.gt.2019-10-23T06:23:01Z` returns all `Datasets` in a
+          space created after that date.'
+        required: false
+        type: string
+      - name: order
+        in: query
+        description: 'With the `order_by` param, returns all `Dataset`s in a `Space`
```
Comment (amirbey) [inline swagger.yaml]:
```suggestion
        description: 'With the `order` param, returns all `Dataset`s in a `Space`
```

File: swagger.yaml [hunk 5]
```diff
@@ -129,6 +129,108 @@ paths:
...
+          ordered in `asc` or `desc` order. Example: `?order=asc&order_by=created_at`'
+        required: false
+        type: string
+      - name: order_by
+        in: query
+        description: 'With the `order` param, returns all `Dataset`s in a `Space`
```
Comment (amirbey) [inline swagger.yaml]:
```suggestion
        description: 'With the `order_by` param, returns all `Dataset`s in a `Space`
```

File: swagger.yaml [hunk 6]
```diff
@@ -129,6 +129,108 @@ paths:
...
+      - name: data_source
+        in: path
+        description: "`DataSource` token"
+        required: true
+        type: string
+      - name: filter
+        in: query
+        description: 'Returns a filtered list of `Dataset`s in a `Space` filtered
+          by whether their `created_at` or `updated_at` timestamps are `gt` (greater
+          than) or `lt` (less than) the datetime passed in (ISO8601 format). Example:
```
Comment (amirbey) [inline swagger.yaml]:
```suggestion
          than) or `lt` (less than) the datetime passed in ISO8601 format. Example:
```

**Review by amirbey — COMMENTED** [review]
A few things suggestions below.

Also a bit of a personal preference, but I'd prefer _order_by_ to appear before _order_ for parameter optinos.

File: swagger.yaml (hunk 5 above)
Comment (marianmosley) [reply swagger.yaml]: Thanks for catching this!  The same issue existed for reports.
image

**Comment by marianmosley** [comment]
@amirbey Thanks so much for catching these errors!
 I am going to leave the order/order_by, um, _order_ for now so that it's consistent with reports.

**Review by msalli — APPROVED** [review]
👍

**Review by monteverdetico — APPROVED** [review]

**Pull request merged**